File: app/routes/decisions/schemas.py
from datetime import datetime
from typing import List
from uuid import UUID

from pydantic import BaseModel

from app.routes.utils import ORMModel, MoneyAmountSchema
from app.routes.catalog.schemas import SKUWithProductResponseSchema
from core.models.decisions import Decision


class BuyDecisionResponseSchema(ORMModel):
    id: UUID
    sku: SKUWithProductResponseSchema
    decision: Decision
    quantity: int
    buy_vwap: MoneyAmountSchema
    expected_resale_net: MoneyAmountSchema
    asof_listings: datetime
    asof_sales: datetime
    reason_codes: List[str]
    created_at: datetime


class BuyDecisionsResponseSchema(BaseModel):
    decisions: List[BuyDecisionResponseSchema]
    total_count: int
    filters_applied: dict

File: web/app/buy-decisions/schemas.ts
import { z } from "zod";
import { SKUWithProductResponseSchema } from "../catalog/schemas";
import { MoneyAmountSchema } from "../schemas";

export const MarketplaceSchema = z.enum(["TCGPLAYER"]);

export const DecisionSchema = z.enum(["BUY", "PASS"]);

export const BuyDecisionResponseSchema = z.object({
  id: z.string(),
  sku: SKUWithProductResponseSchema,
  decision: DecisionSchema,
  quantity: z.number(),
  buy_vwap: MoneyAmountSchema,
  expected_resale_net: MoneyAmountSchema,
  asof_listings: z.string().transform((val) => new Date(val)),
  asof_sales: z.string().transform((val) => new Date(val)),
  reason_codes: z.array(z.string()),
  created_at: z.string().transform((val) => new Date(val)),
});

export const BuyDecisionsResponseSchema = z.object({
  decisions: z.array(BuyDecisionResponseSchema),
  total_count: z.number(),
  filters_applied: z.record(z.any()),
});

export type BuyDecisionResponse = z.infer<typeof BuyDecisionResponseSchema>;
export type BuyDecisionsResponse = z.infer<typeof BuyDecisionsResponseSchema>;
